### PR TITLE
commands: use ReaderWithStats in flashback_to_version command

### PR DESCRIPTION
**Pull request by JmPotato**
Signed-off-by: JmPotato <ghzpotato@gmail.com>

### What is changed and how it works?

Issue Number: ref #13303.

What's Changed:

```commit-message
Use `ReaderWithStats` in `flashback_to_version` command to collect the statistics info as much as possible.
```

### Check List 

Tests 

- Unit test

### Release note 

```release-note
None.
```

### CODE REVIEW

File: src/storage/txn/actions/flashback_to_version.rs [hunk 1]
```diff
@@ -191,7 +197,10 @@ pub mod tests {
         let cm = ConcurrencyManager::new(TimeStamp::zero());
         let mut txn = MvccTxn::new(start_ts, cm);
         let snapshot = engine.snapshot(Default::default()).unwrap();
-        let mut reader = SnapshotReader::new_with_ctx(version, snapshot, &ctx);
+        let mut reader = ReaderWithStats::new(
+            SnapshotReader::new_with_ctx(version, snapshot, &ctx),
+            &mut statistics,
```
Comment (sticnarf) [inline src/storage/txn/actions/flashback_to_version.rs:202]: Do you want to assert about whether there is something in `statistics` as you're using it in tests?

Comment (JmPotato) [reply src/storage/txn/actions/flashback_to_version.rs]: Not necessarily needed, I removed it first.

**Review by sticnarf — APPROVED** [review]

**Review by Connor1996 — APPROVED** [review]
LGTM

**Pull request merged**